### PR TITLE
PLAT-23628: Add partner level allowed domains

### PR DESCRIPTION
**Pull request by danbar0kal**

### CODE REVIEW

File: admin_console/forms/PartnerConfiguration.php [hunk 1]
```diff
@@ -477,6 +477,11 @@ public function init()
 			'decorators' => array('ViewHelper', array('HtmlTag',  array('tag' => 'dt', 'id' => 'mothly_bandwidth_combined')))
 		));
 		
+		$this->addElement('text', 'partner_level_allowed_domains', array(
```
Comment (inbal-ben-david) [inline admin_console/forms/PartnerConfiguration.php:480]: can be just allowed_domaind as the whole configuration is partner configuration
also move it to the section where we have 'CDN Host white list regex' as this new option is not related to 'Included Usage' section

File: alpha/lib/model/Partner.php [hunk 2]
```diff
@@ -99,6 +99,8 @@ class Partner extends BasePartner
 	
 	const EXCLUDED_ADMIN_ROLE_NAME = 'excluded_admin_role_name';
 	
+	const PARTNER_LEVEL_ALLOWED_DOMAINS = 'partner_level_allowed_domains';
```
Comment (inbal-ben-david) [inline alpha/lib/model/Partner.php:102]: can be just allowedDomains on all functions

File: alpha/lib/model/Partner.php [hunk 3]
```diff
@@ -1130,9 +1132,11 @@ public function setTemplateCustomMetadataNum($v)	{$this->putInCustomData('templa
 	public function setInitialPasswordSet($v)			{$this->putInCustomData('initial_password_set', $v);}
 	public function setMarketoCampaignId($v)			{$this->putInCustomData('marketo_campaign_id', $v);}
 	public function setExcludedAdminRoleName($v)			{$this->putInCustomData(self::EXCLUDED_ADMIN_ROLE_NAME, $v);}
+	public function setPartnerLevelAllowedDomains($v)		{$this->putInCustomData(self::PARTNER_LEVEL_ALLOWED_DOMAINS,$v);}
 
 	public function getLoginUsersQuota()				{return $this->getFromCustomData('login_users_quota', null, 0);}
 	public function getExcludedAdminRoleName()			{return $this->getFromCustomData(self::EXCLUDED_ADMIN_ROLE_NAME, null, 0);}
+	public function getPartnerLevelAllowedDomains()		{return $this->getFromCustomData(self::PARTNER_LEVEL_ALLOWED_DOMAINS,null,0);}
```
Comment (inbal-ben-david) [inline alpha/lib/model/Partner.php:1139]: indentation

File: plugins/admin_console/system_partner/lib/api/KalturaSystemPartnerConfiguration.php [hunk 4]
```diff
@@ -832,6 +838,14 @@ public function toObject ( $object_to_fill = null , $props_to_skip = array() )
 		{
 			$object_to_fill->setExcludedAdminRoleName(null);
 		}
+		if(!is_null($this->partnerLevelAllowedDomains))
+		{
+			$object_to_fill->setPartnerLevelAllowedDomains($this->partnerLevelAllowedDomains);
+		}
+		else
+		{
+			$object_to_fill->setPartnerLevelAllowedDomains(null);//should be handled like fix in PLAT-23622
```
Comment (inbal-ben-david) [inline plugins/admin_console/system_partner/lib/api/KalturaSystemPartnerConfiguration.php:847]: lets remove the comment as we are going to handle this

File: admin_console/forms/PartnerConfiguration.php [hunk 5]
```diff
@@ -1089,7 +1096,10 @@ public function addAllDisplayGroups($permissionNames)
 									Kaltura_Client_SystemPartner_Enum_SystemPartnerLimitType::ACCESS_CONTROLS.'_max',
 									Kaltura_Client_SystemPartner_Enum_SystemPartnerLimitType::LIVE_STREAM_INPUTS.'_max',
 									Kaltura_Client_SystemPartner_Enum_SystemPartnerLimitType::LIVE_STREAM_OUTPUTS.'_max',
-									Kaltura_Client_SystemPartner_Enum_SystemPartnerLimitType::LIVE_RTC_STREAM_INPUTS.'_max')
+									Kaltura_Client_SystemPartner_Enum_SystemPartnerLimitType::LIVE_RTC_STREAM_INPUTS.'_max',
+									'partner_level_allowed_domains',
```
Comment (inbal-ben-david) [inline admin_console/forms/PartnerConfiguration.php:1100]: why the following are under LIVE_RTC_STREAM_INPUTS?

Comment (danbar0kal) [reply admin_console/forms/PartnerConfiguration.php:1100]: it was more visually pleasing

**Review by inbal-ben-david — APPROVED** [review]
